package.json

```diff
@@ -46,6 +46,7 @@
   "dependencies": {
     "history": "^3.0.0",
     "lodash.assign": "^4.2.0",
+    "lodash.isequal": "^4.4.0",
```

Comment:
Oh ya - this is fine for my use case, but just to check first: are `state.query` and `action.payload.query` the same memory object in the no-op case you're trying to catch here?

If so, then there isn't even a reason to shallow compare, as we could simply use `===`.

On the other hand, if we want to no-op when they are *equivalent*, then you may in fact wish to use a deep equivalency function instead of a shallow one: I'm not 100% sure how this parses here, but typically the search value `?foo=a&foo=b` would parse to `{foo: ['a', 'b']}`. If this is true, then something like lodash's `isEqual` is required for the desired behavior, since `['a','b'] !== ['a','b']`.